augur/api/view/routes.py

```diff
@@ -48,163 +60,126 @@ def logo(brand=None):
-    about the current status of collection in the backend
+    This renders a status/loading page.
 """
 @app.route('/collection/status')
 def status_view():
-    return render_module("status", title="Status")
+    return render_module("status")
 
 """ ----------------------------------------------------------------
-login:
-    Under development
+    This route performs user authentication.
 """
 @app.route('/account/login', methods=['GET', 'POST'])
 def user_login():
+    # redirect users who are already logged in
+    if current_user.is_authenticated:
+        return redirect(request.args.get('next', url_for('user_settings')))
+    
     if request.method == 'POST':
+        username = request.form.get('username', None)
+        password = request.form.get('password', None)
+        
+        if not username or not password:
+            flash("Please provide both a username and password.", "error")
+            return render_template("login.j2")
+        
         try:
-            username = request.form.get('username')
-            remember = request.form.get('remember') is not None
-            password = request.form.get('password')
-            register = request.form.get('register')
-
-            if username is None:
-                raise LoginException("A login issue occurred")
-
-            user = User.get_user(db_session, username)
-
-            if not user and register is None:
-                raise LoginException("Invalid login credentials")
+            user = User.get_by_login(db_session, username)
+            if not user:
+                raise LoginException("Invalid username or password")
             
-            # register a user
-            if register is not None:
-                if user:
-                    raise LoginException("User already exists")
-                
-                email = request.form.get('email')
-                first_name = request.form.get('first_name')
-                last_name = request.form.get('last_name')
-                admin = request.form.get('admin') or False
-
-                result = User.create_user(username, password, email, first_name, last_name, admin)
-                if not result[0]:
-                    raise LoginException("An error occurred registering your account")
-                else:
-                    user = User.get_user(db_session, username)
-                    flash(result[1]["status"])
-
-            # Log the user in if the password is valid
-            if user.validate(password) and login_user(user, remember = remember):
-                flash(f"Welcome, {username}!")
-                if "login_next" in session:
-                    return redirect(session.pop("login_next"))
-                return redirect(url_for('root'))
-            else:
-                print("Invalid login")
-                raise LoginException("Invalid login credentials")
+            if not user.check_password(password):
```

Comment:
There is no such function on the User class called `check_password`, and that file was not updated in this PR to add it.